client/client.cpp

```diff
@@ -0,0 +1,69 @@
+#include "client.h"
+#include <arpa/inet.h>
+#include <csignal>
+#include <cstring>
+#include <errno.h>
+#include <iostream>
+#include <netinet/in.h>
+#include <regex>
+#include <sys/socket.h>
+#include <unistd.h>
+
+int Client::serverfd = 0;
+bool Client::killCalled = false;
+
+int Client::server_connect(const char *pIp, const char *pPort) {
+  int clientSock = socket(PF_INET, SOCK_STREAM, 0);
+
+  if (clientSock < 0) {
+    std::cout << "Failed to create socket: " << strerror(errno) << std::endl;
+    return -1;
+  }
+
+  sockaddr_in *serverAddr;
+  socklen_t socklen = sizeof(struct sockaddr);
+  memset(serverAddr, 0, socklen);
+
+  serverAddr->sin_family = AF_INET;
+  serverAddr->sin_port = htons(atoi(pPort));
+
+  if (inet_pton(AF_INET, pIp, &serverAddr->sin_addr) <= 0) {
+    std::cout << "Failed to convert Ip to network: " << strerror(errno)
+              << std::endl;
+    return -1;
+  }
+
+  int err = connect(clientSock, (struct sockaddr *)serverAddr, socklen);
+  if (err < 0) {
+    std::cout << "Failed to Connect: " << strerror(errno) << std::endl;
+    return -1;
+  }
+  serverfd = clientSock;
+
+  return serverfd;
+}
+
+int Client::checkIp(const char *pIp) {
+  const std::regex rIp("^\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}$");
+
+  if (!std::regex_search(pIp, rIp)) {
+  std:;
+    std::cout << "IP format is incorrect. Format => 255.255.255.255"
+              << std::endl;
+    return -1;
+  }
+
+  return 0;
+}
+
+int Client::checkPort(const char *pPort) {
```

Comment:
Better if we check the provided parameters (IP and Port) outside the client code. The client needs to be called with two already-prepared values: 16-bit int port and 32-bit int address. It is ok if the numbers are still staying in host representation rather than network. Check the main.cpp for the server